pkg/async/schedule/flytescheduler/interfaces/go_gf_wrapper.go

```diff
@@ -0,0 +1,19 @@
+package interfaces
+
+import (
+	"context"
+	"github.com/flyteorg/flyteadmin/pkg/repositories/models"
+	"time"
+)
+
+// GoGFWrapper Wrapper interface to the gogf framework which is used for scheduler
+type GoGFWrapper interface {
+	// Register add and remove cron or fixed rate schedules from gogf
+	Register(ctx context.Context, s models.SchedulableEntity, funcRef func()) error
```

Comment:
Shouldnt the funcRef be of the format
```
funcRef func(ctx context.Context, scheduleTime time.Time) error
```
This makes it the most future proof. Context is always a good thing to add, this way we can allow graceful termination. Also schedule_time should be handled in our wrapper, Ideally the cron system should have done it, but sadly we have to. And we can have a common way of handling errors.

Reply:
Can we discuss more over this on chat.
the semantic of the func is that it would run a catch up between last execution time and the current triggered time of the job func.
So it can be not just one schedule time but a list of schedule times that this one func can be doing.